src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java

```diff
@@ -180,6 +183,19 @@ public void construct() {
     queueHealthDecider.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
     queueHealthDecider.addAllUpstreams(Collections.singletonList(queueRejectionClusterRca));
 
+    // Node Config Collector
+    ThreadPool_QueueCapacity queueCapacity = new ThreadPool_QueueCapacity();
+    queueCapacity.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(queueCapacity);
+
+    Cache_Max_Size cacheMaxSize =  new Cache_Max_Size(EVALUATION_INTERVAL_SECONDS);
+    cacheMaxSize.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(cacheMaxSize);
+
+    NodeConfigCollector nodeConfigCollector = new NodeConfigCollector(RCA_PERIOD, queueCapacity, cacheMaxSize);
+    nodeConfigCollector.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    nodeConfigCollector.addAllUpstreams(Arrays.asList(threadpool_RejectedReqs, cacheMaxSize));
+
```

Comment:
can we also add the node config cluster RCA altogether ?

Reply:
Sure, let me add the Cluster RCA as well.